text/0001-rfc-process.md

```diff
@@ -0,0 +1,428 @@
+    to update your `0000-` prefix to that number.
+  - Build consensus and integrate feedback. RFCs that have broad support are
+    much more likely to make progress than those that don't receive any
+    comments. Feel free to reach out in particular to get help identifying
+    stakeholders and obstacles.
+  - The team will discuss the RFC pull request, as much as possible in the
+    comment thread of the pull request itself. Offline discussion will be
+    summarized on the pull request comment thread.
+  - RFCs rarely go through this process unchanged, especially as alternatives
+    and drawbacks are shown. You can make edits, big and small, to the RFC to
+    clarify or change the design, but make changes as new commits to the pull
+    request, and leave a comment on the pull request explaining your changes.
+    Specifically, do not squash or rebase commits after they are visible on the
+    pull request.
+  - At some point, a member of the team will propose a "motion for final
+    comment period" (FCP), along with a *disposition* for the RFC (merge, close,
+    or postpone).
+    - This step is taken when enough of the tradeoffs have been discussed that
+      the core team is in a position to make a decision. That does not require
+      consensus amongst all participants in the RFC thread (which is usually
+      impossible). However, the argument supporting the disposition on the RFC
+      needs to have already been clearly articulated, and there should not be a
+      strong consensus *against* that position. Team members use their best
+      judgment in taking this step, and the FCP itself ensures there is ample
+      time and notification for stakeholders to push back if it is made
+      prematurely.
+    - For RFCs with lengthy discussion, the motion to FCP is usually preceded by
+      a *summary comment* trying to lay out the current state of the discussion
+      and major tradeoffs/points of disagreement.
+    - The motion for FCP should include a checkbox list that tags major
+      stakeholder and reviewers that need to sign off on the FCP. This is often
+      the point at which many team members first review the RFC in full depth.
+      Once a majority of reviewers approve (and at most 2 approvals are outstanding),
+      the FCP begins.
+  - The FCP lasts ten calendar days, so that it is open for at least 5 business
+    days. It is also advertised widely,
+    e.g. on the LF mailing list. This way all
+    stakeholders have a chance to lodge any final objections before a decision
+    is reached.
+  - In most cases, the FCP period is quiet, and the RFC is either merged or
+    closed. However, sometimes substantial new arguments or ideas are raised,
+    the FCP is canceled, and the RFC goes back into development mode.
+
+## The RFC life-cycle
+[The RFC life-cycle]: #the-rfc-life-cycle
+
+Once an RFC becomes "active" then authors may implement it and submit the
```

Comment:
Maybe the sentence is formulated a bit too strict here. Of course, a contributor may also work on an implementation earlier on. It would be risky, however. The main goal of RFCs as proposed here is to build an agreement behind a proposal, not behind an implementation. Also, it acts as a mechanism for contributors to get feedback early on so that they can understand if the feature they are planning is desirable before hey put in the work of implementing it. I think this goal is likely different from the IETF RFCs.

Reply:
I think developing them side by side is a good feasibility test and avoids situations where proposals look good on paper but not in reality.

Reply:
I think I agree with Christian here. I think our current workflow, with thorough code reviews and CI, works quite well for guiding the development of features. But what lacks is a structured way of agreeing on what features we should target (which is exactly what the RFC process is about).

An example is the work I am doing on the [physical clock abstraction in the C runtime]( https://github.com/lf-lang/reactor-c/pull/346), the question of how the LF runtime system should react if a clock synchronization client running on the system, like `chrony` or `ptpd` steps the clocks back or forward 100ms is not only a problem for the C target. I think such questions would be great to discuss through an RFC at the "idea-level"

Reply:
> Of course, a contributor may also work on an implementation earlier on. It would be risky, however.

I agree with this. I think it is important for anyone who works on an implementation before the RFC is approved to be aware that there is a significant risk that their work will be discarded.

> I think such questions would be great to discuss through an RFC at the "idea-level"

I also agree with this, I think that we should have high-level discussions that have the potential to change or reject a proposed feature and that don't get bogged down in details about the implementation.

> I think developing them side by side is a good feasibility test and avoids situations where proposals look good on paper but not in reality.

But on the other hand I also do agree with this. I do not know about anyone else, but if I were to propose a feature, I might be much more comfortable discussing it if I already had a prototype or partial implementation. I think it makes sense to tolerate either approach (implement before RFC approval, or wait for approval) as long as everyone understands the risk of rejecting a premature implementation and as long as people agree to focus RFCs on high-level discussion